### PR TITLE
Add a note to the contributing docs that joining the mailing list gives access to build logs

### PR DESCRIPTION
**Pull request by roberthbailey**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind breaking
> /kind bug
> /kind cleanup
/kind documentation
> /kind feature
> /kind hotfix

**What this PR does / Why we need it**: Updates the contributing docs, adding info for new contrubitors.

**Which issue(s) this PR fixes**:

Closes #

**Special notes for your reviewer**:

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -123,4 +124,4 @@ repository, we have a [community membership guide](./docs/governance/community_m
   - Info - should be used to document state changes in the application or some entity within the application. These logs provide the skeleton of what has happened. Examples: system started/stopped, remote API calls, a new user has been created/updated, etc.
   - Debug - diagnostic information goes here and everything that can help to troubleshoot an application. Examples: any values in business logic, detailed information about the data flow.
 
-  More details can be found here: https://reflectoring.io/logging-levels/
\ No newline at end of file
+  More details can be found here: https://reflectoring.io/logging-levels/
```
Comment (aLekSer) [inline CONTRIBUTING.md:127]: This link is not accessible:
 https://reflectoring.io/logging-levels/
ERR_HTTP2_PROTOCOL_ERROR

Comment (aLekSer) [reply CONTRIBUTING.md:127]: It could be a copy of this blog post:
https://medium.com/@tom.hombergs/tip-use-logging-levels-consistently-913b7b8e9782

Comment (roberthbailey) [reply CONTRIBUTING.md:127]: The link works for me, but comparing to the medium article they look identical.

Comment (roberthbailey) [reply CONTRIBUTING.md]: Pushed a commit with the medium link.

**Pull request merged**